csharp_dotnetcore/8.Suggested-Actions/Suggested_Actions/SuggestedActionsBot.cs

```diff
@@ -0,0 +1,121 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+using Microsoft.Bot.Builder;
+using Microsoft.Bot.Schema;
+
+namespace Suggested_Actions
+{
+    /// <summary>
+    /// This bot will respond to the user's input suggested actions.
+    /// </summary>
+    public class SuggestedActionsBot : IBot
+    {
+        /// <summary>
+        /// This controls what happens when an activity gets sent to the bot.
+        /// </summary>
+        /// <param name="turnContext">Provides the context for the turn of the bot.</param>
+        /// <param name="cancellationToken">A cancellation token that can be used by other objects
+        /// or threads to receive notice of cancellation.</param>
+        /// <returns>A task.</returns>
+        public async Task OnTurnAsync(ITurnContext turnContext, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            if (turnContext == null)
+            {
+                throw new ArgumentNullException(nameof(turnContext));
+            }
+
+            switch (turnContext.Activity.Type)
+            {
+                case ActivityTypes.Message:
+                    var text = turnContext.Activity.Text.ToLowerInvariant();
+                    switch (text)
```

Comment:
I'm not a huge fan of these nested switch statements. Consider making this a method to be more clear.